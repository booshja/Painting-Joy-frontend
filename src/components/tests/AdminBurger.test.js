import { render } from "@testing-library/react";
import AdminBurger from "../AdminBurger";
import MenuContext from "../../context/MenuContext";

/** Smoke Test */
it("renders without crashing, with context", () => {
    render(
        <MenuContext.Provider value={{ menuOpen: true }}>
            <AdminBurger />
        </MenuContext.Provider>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <MenuContext.Provider value={{ menuOpen: true }}>
            <AdminBurger />
        </MenuContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByTestId } = render(
        <MenuContext.Provider value={{ menuOpen: true }}>
            <AdminBurger />
        </MenuContext.Provider>
    );
    const div1 = getByTestId("div1");
    const div2 = getByTestId("div2");
    const div3 = getByTestId("div3");

    expect(div1).toBeInTheDocument();
    expect(div2).toBeInTheDocument();
    expect(div3).toBeInTheDocument();
});
