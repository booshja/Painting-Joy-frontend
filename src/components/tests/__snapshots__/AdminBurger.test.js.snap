// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM czyvtI"
    open=""
  >
    <div
      data-testid="div1"
    />
    <div
      data-testid="div2"
    />
    <div
      data-testid="div3"
    />
  </button>
</DocumentFragment>
`;
